src/services/db/GitFileSystemService.ts

```diff
@@ -112,24 +113,58 @@ export default class GitFileSystemService {
       })
   }
 
-  // Obtain the Git blob hash of a file or directory
-  getGitBlobHash(
-    repoName: string,
-    filePath: string
-  ): ResultAsync<string, GitFileSystemError> {
+  // Ensure that the repository is in the BRANCH_REF branch
+  ensureCorrectBranch(repoName: string): ResultAsync<true, GitFileSystemError> {
     return ResultAsync.fromPromise(
       this.git
         .cwd(`${EFS_VOL_PATH}/${repoName}`)
-        .checkout(BRANCH_REF)
-        .revparse([`HEAD:${filePath}`]),
+        .revparse(["--abbrev-ref", "HEAD"]),
       (error) => {
         if (error instanceof GitError) {
           return new GitFileSystemError(error.message)
         }
 
-        logger.error(`Error when getting Git blob hash: ${error}`)
+        logger.error(`Error when getting current branch: ${error}`)
         return new GitFileSystemError("An unknown error occurred")
       }
+    ).andThen((currentBranch) => {
+      if (currentBranch !== BRANCH_REF) {
+        return ResultAsync.fromPromise(
+          this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).checkout(BRANCH_REF),
+          (error) => {
+            if (error instanceof GitError) {
+              return new GitFileSystemError(error.message)
+            }
+
+            logger.error(`Error when checking out ${BRANCH_REF}: ${error}`)
+            return new GitFileSystemError("An unknown error occurred")
+          }
+        ).andThen(() => okAsync<true, never>(true))
```

Comment:
[nit] moving forward, we can just `okAsync<true>(true)`

Reply:
Done in bd30514d4857b761be2ddef3f4ba2db0cc6b2188!